### ISSUE TITLE
The configured Repository could not be loaded.

### ISSUE THREAD

**Issue by barneco** [auth0/laravel-auth0#353]
### SDK Version

7.5

### PHP Version

PHP 8.1

### Composer Version

2.x

### What happened?

Upgraded to 7.5 today and am now receiving an error.  I reviewed the upgrade guide and didn't see anything about required changes to a custom repo.

Error:

`The configured Repository App\Repositories\CustomUserRepository could not be loaded.`

This custom repo worked prior to 7.5 and no changes have been made.

my auth.php config:
```
    'guards' => [

        'auth0' => [
            'driver' => 'auth0.guard',
            'provider' => 'auth0',
        ],

```
```
     'providers' => [
        /*'users' => [
            'driver' => 'eloquent',
            'model' => App\Models\User::class,
        ],*/
        'users' => [
            'driver' => 'auth0',
            'model' => App\Models\User::class,
        ],

        'auth0' => [
            'driver' => 'auth0.provider',
            'repository' => \App\Repositories\CustomUserRepository::class,
        ],
```

```
<?php

namespace App\Repositories;

use App\Http\Controllers\Auth\Auth0IndexController;
use App\Models\Role;
use App\Models\User;

use Auth0\SDK\Exception\StateException;
use Illuminate\Contracts\Auth\Authenticatable;

class CustomUserRepository implements \Auth0\Laravel\Contract\Auth\User\Repository
{
    /**
     * Generate a \Auth0\Laravel\Model\Stateful\User instance from an available Auth0-PHP user session.
     *
     * @param array $user An array containing the raw Auth0 user data.
     */
    public function fromSession(array $user): ?\Illuminate\Contracts\Auth\Authenticatable
    {
        ...
    }

    /**
     * Generate a \Auth0\Laravel\Model\Stateful\User instance from a parsed Access Token.
     *
     * @param array $user An array containing the raw Auth0 user data.
     */
    public function fromAccessToken(array $user): ?\Illuminate\Contracts\Auth\Authenticatable
    {
        return null;
    }

    ...
}
```

### How can we reproduce this issue?

implement auth0 solution using a custom user repository as described in the quick start docs and use v7.5 of the sdk.  attempt to login to application.

**Comment by ngmy** [comment]
I encountered the same error.

I don't know if this is the right way to do it, but I registered the repository in the service container using the `AppServiceProvider::register` method, etc. and the error stopped appearing for now.

```php
use My\Repository;

$this->app->singleton(Repository::class, static fn (): Repository => new Repository());
```

**Comment by ABartelt** [comment]
Don't you have to change the following:
```
....
'guards' => [

        'auth0' => [
            'driver' => 'auth0.guard',
            'provider' => 'auth0.provider',
        ],
...
```

as it mentioned in the [release notes of version 7.5.0](https://github.com/auth0/laravel-auth0/releases/tag/7.5.0) ?

**Comment by barneco** [comment]
> Don't you have to change the following:
> 
> ```
> ....
> 'guards' => [
> 
>         'auth0' => [
>             'driver' => 'auth0.guard',
>             'provider' => 'auth0.provider',
>         ],
> ...
> ```
> 
> as it mentioned in the [release notes of version 7.5.0](https://github.com/auth0/laravel-auth0/releases/tag/7.5.0) ?

yep, take a look at the code in the [issue details](https://github.com/auth0/laravel-auth0/issues/353#issue-1652852622).  i'm using the new driver names.

also, FYI, your code snippet is slightly incorrect.  you need to specify 'auth0.provider' as the **provider driver**, not the guard provider.  again, reference the[ issue details](https://github.com/auth0/laravel-auth0/issues/353#issue-1652852622)

**Comment by evansims** [comment]
Hey @barneco, all 👋 Sorry to see you're hitting issues with the update! Give me a few to investigate and see if I can reproduce the issue here, and I'll circle back to you here.

**Comment by evansims** [comment]
I believe I've identified the issue; @barneco could you try updating to the main branch (`dev-main`) and see if that resolves things for you?

**Comment by barneco** [comment]
@evansims That did the trick, thanks for the prompt response.  I'm now getting a weird redirect loop after logging in, but I'm no longer receiving the error regarding the custom user repo, so I think that's fixed.  I'll keep looking and open another issue if I can't find the root cause on my side.

**Comment by evansims** [comment]
@barneco No worries, glad that fixed things for you! Apologies for the trouble. I'll get a patch release shipped to resolve it yet today. Thanks for letting me know about the issue.

Please do let me know if you continue having other issues, happy to get things sorted.

**Comment by ABartelt** [comment]
> > Don't you have to change the following:
> > ```
> > ....
> > 'guards' => [
> > 
> >         'auth0' => [
> >             'driver' => 'auth0.guard',
> >             'provider' => 'auth0.provider',
> >         ],
> > ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > as it mentioned in the [release notes of version 7.5.0](https://github.com/auth0/laravel-auth0/releases/tag/7.5.0) ?
> 
> yep, take a look at the code in the [issue details](https://github.com/auth0/laravel-auth0/issues/353#issue-1652852622). i'm using the new driver names.
> 
> also, FYI, your code snippet is slightly incorrect. you need to specify 'auth0.provider' as the **provider driver**, not the guard provider. again, reference the[ issue details](https://github.com/auth0/laravel-auth0/issues/353#issue-1652852622)

Ups, your right, sorry!
